pilot/pkg/model/push_context.go

```diff
@@ -1459,6 +1463,101 @@ func (ps *PushContext) initServiceRegistry(env *Environment) {
 	ps.initServiceAccounts(env, allServices)
 }
 
+// resolveServiceAliases sets the Aliases attributes on all services. The incoming Service's will just have AliasFor set,
+// but in our usage we often need the opposite: for a given service, what are all the aliases?
+// resolveServiceAliases walks this 'graph' of services and updates the Alias field in-place.
+func resolveServiceAliases(allServices []*Service, configsUpdated sets.Set[ConfigKey]) {
```

Comment:
can you optimize this function, it is very confusing and have 6 loops

Reply:
I spent a long time on this, its extensively commented and is as simple as I was able to make it. Can you give more direct suggestions? I cannot improve it any further without help.

Reply:
I added a bit more comments, but spent a while and couldn't make the code any simpler.